src/group_actions/group_action_interface.jl

```diff
@@ -85,41 +85,47 @@ $(_note_action_argument_order)
 """
 abstract type AbstractRightGroupActionType <: AbstractGroupActionType end
 
-@doc """
-    AbstractGroupAction{T<:AbstractGroupActionType, L<:LieGroup, M<:AbstractManifold}
+"""
+    GroupAction{T<:GroupActionType, L<:LieGroup, M<:AbstractManifold}
 
-An abstract group action of [`AbstractGroupActionType`](@ref) `T` of a [`LieGroup`](@ref) of type `L`
-acting on an $(_link(:AbstractManifold)) of type `M`.
+Specify a group action of [`AbstractGroupActionType`](@ref) `T` of a [`LieGroup`](@ref) `G` acting on `M`.
 
-See [HilgertNeeb:2012; Section 9.1.3](@cite) for more details.
-"""
-abstract type AbstractGroupAction{
-    T<:AbstractGroupActionType,L<:LieGroup,M<:AbstractManifold
-} end
+Let ``$(_math(:M))`` be a $(_link(:AbstractManifold)) and ``$(_math(:G))`` be a [`LieGroup`](@ref) woth group operation ``$(_math(:∘))``.
 
-function base_lie_group end
-@doc """
-    base_lie_group(A::AbstractGroupAction)
+A (smooth) action of the group ``$(_math(:G))`` on the manifold ``$(_math(:M))`` is a map
 
-Return the [`LieGroup`](@ref) of the [`AbstractGroupAction`](@ref)
-specifying the action.
-"""
-base_lie_group(::AbstractGroupAction)
+```math
+σ: $(_math(:G)) × $(_math(:M)) → $(_math(:M))
+```
 
-@doc """
-    base_manifold(A::AbstractGroupAction)
+with the properties
+
+* ``σ($(_math(:e)), p) = p`` holds for all ``p ∈ $(_math(:M))``
+* ``σ(g, σ(h, p)) = σ(g$(_math(:∘))h, p)`` holds for all ``g,h ∈ $(_math(:G))``, ``p ∈ $(_math(:M))``
+
+
+# Fields
+
+* `type::T`: The type of the group action.
+* `group::L`: The group acting.
+* `manifold::M`: The manifold the group acts upon.
+
+See [HilgertNeeb:2012; Section 9.1.3](@cite) for more details.
 
-Return the $(_link(:AbstractManifold)) the group action acts upon.
 """
-ManifoldsBase.base_manifold(::AbstractGroupAction)
+struct GroupAction{T<:AbstractGroupActionType,L<:LieGroup,M<:ManifoldsBase.AbstractManifold}
```

Comment:
Could you add somewhere the intended way to construct `GroupAction` for group operation action?  Should `manifold` be the Lie group itself or the underlying manifold?

Reply:
I wanted to add a test for `switch(A::GroupAction) = GroupAction(switch(A.type), A.group)` and I wasn't sure what to do there.

Reply:
Nice, for such generic things I indeed wanted to use a few dummies.

Reply:
For the `GroupOperationAction` That should be defined on `G` and not `M`, since we might have Lie groups on the same manifold with different group ops. and it should create a `GroupAction(GroupOperationActionType, G,G)`.

Reply:
For the docs, it is documented at https://juliamanifolds.github.io/LieGroups.jl/previews/PR14/interface/actions/#LieGroups.GroupOperationAction-Tuple{AbstractGroupActionType,%20LieGroup}, but you are right, that could be improved, it also works or the other 3 cases (right, invleft, invright).

Reply:
Thanks, I didn't think to check in that part of documentation.